chat/chat.py

```diff
@@ -1,41 +1,58 @@
 import openai
-from config import AppConfig
 from abc import ABC, abstractmethod
+from config import AppConfig
 
 class Chat(ABC):
     @abstractmethod
     def generate_response(self, query: str, context: str = None) -> str:
-        """Generate a response based on the query and optional context."""
+        """Generate a response based on the query and current session's context."""
         pass
 
 class ChatOpenAI(Chat):
-    
+
     def __init__(self):
         config = AppConfig()
         openai.api_key = config.openai_api_key
-        self.history = []
+        self.history = [self._get_generate_system()]
+
+    def _get_generate_system(self) -> dict:
 
-    def generate_response(self, query, context=None):
+        return {
+            "role": "system",
+            "content": """
+                        You are an assistant that can only provide responses based on the PROVIDED CONTEXT. 
+                        DO NOT USE AN EXTERNAL/GENERAL KNOWLEDGE, only answer based on PROVIDED CONTEXT. !!!
+                        """
+        }
+
+    def generate_response(self, query: str, context: str = None) -> str:
         if not query:
-            return ""  
-        
+            return ""
+
         if context:
-            full_input = f"Given a context: {context}\n Given a query: {query}\n Please answer query based on the given context"
+            full_input = f"Given a context: {context}\n Given a query: {query}\n Please answer query based on the given context."
         else:
             full_input = query
-        
+
         self.history.append({"role": "user", "content": full_input})
-        
+
         response = openai.ChatCompletion.create(
             model="gpt-4o-mini",
             messages=self.history,
+            temperature = 0.2
```

Comment:
I'm curious, any reason for setting the temperature instead of leaving it at default?